spec/controllers/settings/applications_controller_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'rails_helper'
+
+describe Settings::ApplicationsController do
+  render_views
+  
+  let(:user) { Fabricate(:user) }
+  let(:app) { Fabricate(:application, owner: user) }
+  
+  before do
+    sign_in user, scope: :user
+  end
+
+  describe 'GET #index' do
+    let(:other_app) { Fabricate(:application) }
+    before {
+      app
+      other_app
+    }
+
+    it 'shows apps' do
+      get :index
+      expect(response).to have_http_status(:success)
+      expect(assigns(:applications)).to include(app)
+      expect(assigns(:applications)).to_not include(other_app)
+    end
+  end
+
+  
+  describe 'GET #show' do
+    it 'returns http success' do
+      get :show, params: { id: app.id }
+      expect(response).to have_http_status(:success)
```

Comment:
Spec that it acutally shows the application. (testing `assigns(:application)` is enough.)